store.go

```diff
@@ -1,51 +1,70 @@
 package vstore
 
-// Store is the main redux-like store. Use it to access the State properties
-// through the ComponentConstructor.
-type Store struct {
-	r    Reducer
-	mws  []Middleware
-	subs map[*SubFunc]bool
+import (
+	"github.com/gopherjs/vecty"
+	"sync"
+)
+
+// store is the main redux-like store. Use it to access the State properties.
+type Store interface {
+	Connect(comp StoreComponent) StoreComponent
+	Dispatch(action interface{})
+	State() interface{}
 }
 
-// SubFunc is a subscription callback.
-type SubFunc func(s *Store)
+// store satisfies Store.
+type store struct {
+	r   Reducer
+	mws []Middleware
+
+	mtx   sync.RWMutex
```

Comment:
why not a regular `sync.Mutex` since there's no reading or writing in the way we're syncing Dispatch.

Reply:
Subscribe and unsubscribe use the write lock and dispatch uses the read lock while re rerendering. I may make sense to change dispatch to use a write lock on the reduce call.